test/gtest/cfg.cpp

```diff
@@ -82,3 +86,165 @@ TEST(CFGTest, Print) {
 
   EXPECT_EQ(ss.str(), cfgText);
 }
+
+TEST(CFGTest, LinearLiveness) {
+  auto moduleText = R"wasm(
+    (module
+      (func $bar
+        (local $a (i32))
+        (local $b (i32))
+        (local $c (i32))
+        (local.set $a
+          (i32.const 1)
+        )
+        (drop
+          (local.get $a)
+        )
+        (local.set $b
+          (local.get $a)
+        )
+        (local.set $c
+          (i32.const 1)
+        )
+        (drop
+          (local.get $c)
+        )
+      )
+    )
+  )wasm";
+
+  auto analyzerText = R"analyzer(CFG Analyzer
+State Block: 0
+State at beginning: 000
+State at end: 000
+Intermediate States (reverse order): 
+000
+block
+000
+drop
+000
+local.get $2
+100
+local.set $2
+000
+i32.const 1
+000
+local.set $1
+000
+local.get $0
+001
+drop
+001
+local.get $0
+001
+local.set $0
+000
+i32.const 1
+000
+End
+)analyzer";
+
+  Module wasm;
+  SExpressionParser parser(moduleText);
+  SExpressionWasmBuilder builder(wasm, *(*parser.root)[0], IRProfile::Normal);
+
+  CFG cfg = CFG::fromFunction(wasm.getFunction("bar"));
+  const size_t sz = 3;
+  MonotoneCFGAnalyzer<sz> analyzer = MonotoneCFGAnalyzer<sz>::fromCFG(&cfg);
+  analyzer.evaluate();
+
+  bool colors = Colors::isEnabled();
+  Colors::setEnabled(false);
+  std::stringstream ss;
+  analyzer.print(ss);
+  Colors::setEnabled(colors);
+
+  EXPECT_EQ(ss.str(), analyzerText);
+}
+
+TEST(CFGTest, NonlinearLiveness) {
+  auto moduleText = R"wasm(
+    (module
+      (func $bar
+        (local $a (i32))
+        (local $b (i32))
+        (local.set $a
+          (i32.const 1)
+        )
+        (if
+          (i32.eq
+            (local.get $a)
+            (i32.const 2)
+          )
+          (local.set $b
+            (i32.const 4)
+          )
+          (drop
+            (local.get $a)
+          )
+        )
+      )
+    )
+  )wasm";
+
+  auto analyzerText = R"analyzer(CFG Analyzer
+State Block: 0
+State at beginning: 00
+State at end: 01
+Intermediate States (reverse order): 
+01
+i32.eq
+01
+i32.const 2
+01
+local.get $0
+01
+local.set $0
+00
+i32.const 1
+00
+State Block: 1
+State at beginning: 00
+State at end: 00
+Intermediate States (reverse order): 
+00
+local.set $1
+00
+i32.const 4
+00
+State Block: 2
+State at beginning: 01
+State at end: 00
+Intermediate States (reverse order): 
+00
+drop
+00
+local.get $0
+01
+State Block: 3
+State at beginning: 00
+State at end: 00
+Intermediate States (reverse order): 
+00
+block
+00
+End
+)analyzer";
+
+  Module wasm;
+  SExpressionParser parser(moduleText);
+  SExpressionWasmBuilder builder(wasm, *(*parser.root)[0], IRProfile::Normal);
+
+  CFG cfg = CFG::fromFunction(wasm.getFunction("bar"));
+  const size_t sz = 2;
+  MonotoneCFGAnalyzer<sz> analyzer = MonotoneCFGAnalyzer<sz>::fromCFG(&cfg);
+  analyzer.evaluate();
+
+  bool colors = Colors::isEnabled();
+  Colors::setEnabled(false);
+  std::stringstream ss;
+  analyzer.print(ss);
+  Colors::setEnabled(colors);
```

Comment:
Lines `234-236` and `243-247` look repeated in the tests. I assume we'll be adding more in time so the repetition will grow? Perhaps there could be a helper function that does the general outline of things and receives as input the wat text and expected output?

Reply:
We should probably use a [test fixture](https://github.com/google/googletest/blob/main/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests) for this. I'd be happy deferring this to a standalone follow-up.

Reply:
Introduced by #5781. Rebased on it.